### PR TITLE
feat: add GenericTypeValueTransformation for regex-based field extraction

### PR DESCRIPTION
**Pull request by frack113**
## Description

Add a generic transformation to extract fields from values using regex with named groups.

This transformation parses values matching a regex pattern with named capture groups and creates separate detection items for each group.

## Features

- **Named group extraction**: Extract fields using regex named groups `(?P<name>pattern)`
- **Optional field prefix**: Prefix extracted field names with `{field_prefix}.{group_name}`
- **Field filtering**: Restrict transformation to specific fields via `field_to_parse`
- **Type conversion**: Automatic conversion to SigmaType (null, int, float, string)
- **Robust validation**: Comprehensive regex validation (syntax, named groups, duplicates)

### Basic Usage (without field_prefix)
```python
transformation = GenericTypeValueTransformation()
# regex: (?P<type>[A-Za-z0-9_]+):(?P<value>[^\s=|]+)
# Input:  "Dword:00001"
# Output: type: "Dword", value: "00001"
```

### With Field Prefix
```python
transformation = GenericTypeValueTransformation(field_prefix="reg")
# Input:  "Dword:00001"
# Output: reg.type: "Dword", reg.value: "00001"
```

### Custom Regex (domain\username parsing)
```python
transformation = GenericTypeValueTransformation(
    regex=r"(?P<domain>[^\\]*)\\(?P<username>.+)",
    field_prefix="user"
)
# Input:  "contoso\john.doe"
# Output: user.domain: "contoso", user.username: "john.doe"
```

### Empty Domain Edge Case
```python
transformation = GenericTypeValueTransformation(
    regex=r"(?P<domain>[^\\]*)\\(?P<username>.+)",
    field_prefix="user"
)
# Input:  "\john.doe"
# Output: user.username: "john.doe" (empty domain is skipped)
```

### URL Parsing
```python
transformation = GenericTypeValueTransformation(
    regex=r"(?P<protocol>http|https)://(?P<host>[^/]+)(?P<path>/[^?#]*)?",
    field_prefix="url"
)
# Input:  "https://example.com/path/to/resource"
# Output: url.protocol: "https", url.host: "example.com", url.path: "/path/to/resource"
```

### Field Filtering
```python
transformation = GenericTypeValueTransformation(
    field_prefix="reg",
    field_to_parse=["reg", "other_field"]
)
# Only transforms "reg" and "other_field" fields
```

## YAML Configuration

```yaml
processing:
  name: Extract type:value
  priority: 20
  items:
    - id: extract_type_value
      type: generic_type_value
      field_prefix: "reg"
```

## Validation

- Regex must contain at least one named group
- No duplicate named groups allowed
- Comprehensive regex syntax validation
- Empty `field_to_parse` returns `None`

## Tests

19 new tests covering:
- Single/multiple matches
- Custom named groups
- No match scenarios
- Multiple values
- Invalid regex
- Duplicate named groups
- Empty field_prefix (optional)
- Null/number/float value conversion
- Three named groups
- Mixed named/unnamed groups
- Empty domain edge case
- AND condition verification
- Field filtering (field_to_parse)

All 168 tests passing.

## Type Conversion

- `"null"`, `"none"`, or empty string → `SigmaNull`
- Integer strings (without leading zeros) → `SigmaNumber(int)`
- Float strings → `SigmaNumber(float)`
- All other strings → `SigmaString`
- Leading zeros preserved as `SigmaString` (e.g., `"00001"` stays string)

### CODE REVIEW

File: sigma/processing/transformations/values.py [hunk 1]
```diff
@@ -443,3 +443,149 @@ def apply_string_value(self, field: str | None, val: SigmaString) -> SigmaString
             return val.lower()
         else:
             return val.upper()
+
+
+@dataclass
+class GenericTypeValueTransformation(DetectionItemTransformation):
```
Comment (thomaspatzke) [inline sigma/processing/transformations/values.py]: `GenericTypeValueTransformation` is a bit misleading about what the transformation does. What do you think about `ExtractFieldsTransformation`?

File: sigma/processing/transformations/values.py [hunk 2]
```diff
@@ -443,3 +443,149 @@ def apply_string_value(self, field: str | None, val: SigmaString) -> SigmaString
...
+
+    Attributes:
+        regex (str): Regex pattern with named groups (e.g., (?P<name>pattern)).
+            Default: r"(?P<type>[?A-Za-z0-9_]+):(?P<value>[^\\s=|]+)"
+        field_prefix (str | None): Prefix for field names. Used as {field_prefix}.{group_name}.
+            If None, only the group name is used as the field name.
+        field_to_parse (list[str] | None): List of field names to parse. If None, all fields are parsed.
+    """
+
+    regex: str = r"(?P<type>[A-Za-z0-9_]+):(?P<value>[^\s=|]+)"
```
Comment (thomaspatzke) [inline sigma/processing/transformations/values.py]: I would prefer to omit the default regex and make it mandatory to specify one to avoid unexpected side effects if users forget to specify the regex. The default is only one of many use cases for this transformation.

File: sigma/processing/transformations/values.py [hunk 3]
```diff
@@ -443,3 +443,149 @@ def apply_string_value(self, field: str | None, val: SigmaString) -> SigmaString
...
+    field_prefix: str | None = None
+    field_to_parse: list[str] | None = None
```
Comment (thomaspatzke) [inline sigma/processing/transformations/values.py]: This can be specified with the `IncludeFieldCondition` and condition functionality shouldn't be doubled in the transformation to avoid different inconsistent ways to specify field inclusion/exclusion.

File: sigma/processing/transformations/values.py [hunk 4]
```diff
@@ -443,3 +443,149 @@ def apply_string_value(self, field: str | None, val: SigmaString) -> SigmaString
...
+            )
+
+        # Validate no duplicate named groups
+        group_names = list(self.re.groupindex.keys())
+        if len(group_names) != len(set(group_names)):
+            raise SigmaRegularExpressionError(
+                f"Regular expression '{self.regex}' contains duplicate named groups"
+            )
+
+        # Validate regex can match at least an empty string or simple pattern
```
Comment (thomaspatzke) [inline sigma/processing/transformations/values.py]: Don't make any assumptions about the content matched by the regular expression. It can be totally valid that it doesn't matches the empty string.

**Review by thomaspatzke — CHANGES_REQUESTED** [review]

File: sigma/processing/transformations/values.py [hunk 5]
```diff
@@ -443,3 +443,149 @@ def apply_string_value(self, field: str | None, val: SigmaString) -> SigmaString
...
+    - Float strings (without leading zeros) -> SigmaNumber(float)
+    - All other strings -> SigmaString
+
+    Strings with leading zeros (e.g., "00001") are preserved as SigmaString to avoid
+    losing the leading zero information.
+
+    Attributes:
+        regex (str): Regex pattern with named groups (e.g., (?P<name>pattern)).
+            Default: r"(?P<type>[?A-Za-z0-9_]+):(?P<value>[^\\s=|]+)"
+        field_prefix (str | None): Prefix for field names. Used as {field_prefix}.{group_name}.
```
[the remark this answers is not in the document]
Comment (thomaspatzke) [reply sigma/processing/transformations/values.py:484]: Default should be removed anyways.

File: sigma/processing/transformations/values.py [hunk 6]
```diff
@@ -443,3 +443,149 @@ def apply_string_value(self, field: str | None, val: SigmaString) -> SigmaString
...
+                            value=[self._convert_value(group_value)],
+                        )
+                    )
+
+        if new_items:
+            return SigmaDetection(
+                detection_items=new_items,
+                item_linking=ConditionAND,
+            )
+        return None
```
[the remark this answers is not in the document]
Comment (frack113) [reply sigma/processing/transformations/values.py]: Addressed: now returns one nested AND-linked SigmaDetection per value with outer linking following detection_item.value_linking (default OR), and returns None if any value doesn't match (commit 5d9cff2)

File: sigma/processing/transformations/values.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (frack113) [reply sigma/processing/transformations/values.py:484]: Addressed: default regex has been removed entirely (regex is now mandatory) to avoid docstring mismatch (commit 9474e5b)

File: sigma/processing/transformations/values.py [hunk 7]
```diff
@@ -443,3 +443,149 @@ def apply_string_value(self, field: str | None, val: SigmaString) -> SigmaString
...
+        # Don't convert strings with leading zeros (except "0" alone)
+        if value != "0" and value.startswith("0") and value[1:].isdigit():
+            return SigmaString(value)
+        try:
+            return SigmaNumber(int(value))
+        except ValueError:
+            try:
+                return SigmaNumber(float(value))
+            except ValueError:
+                return SigmaString(value)
```
[the remark this answers is not in the document]
Comment (frack113) [reply sigma/processing/transformations/values.py:524]: Addressed: _convert_value now checks value.startswith('0') without the .isdigit() guard, so '03.14' and '00.1' are correctly preserved as SigmaString (commit 9474e5b)

File: tests/test_processing_transformations.py [hunk 8]
```diff
@@ -2912,3 +2912,224 @@ def test_strict_mapped_fields_no_pipeline():
...
+    assert result.detection_items[0].field == "reg.type"
+    assert result.detection_items[0].value == [SigmaString("Dword")]
+    assert result.detection_items[1].field == "reg.value"
+    assert result.detection_items[1].value == [SigmaNumber(1)]
+    # Second value: Qword:2
+    assert result.detection_items[2].field == "reg.type"
+    assert result.detection_items[2].value == [SigmaString("Qword")]
+    assert result.detection_items[3].field == "reg.value"
+    assert result.detection_items[3].value == [SigmaNumber(2)]
+    assert result.item_linking == ConditionAND
```
[the remark this answers is not in the document]
Comment (frack113) [reply tests/test_processing_transformations.py]: Addressed: test updated to expect nested AND-linked detections per value with outer OR linking (commit 9474e5b)

**Review by thomaspatzke — APPROVED** [review]

**Review by thomaspatzke — APPROVED** [review]
